File: file-practice/src/main/kotlin/com/example/filepractice/poc/domain/entity/Order.kt
package com.example.filepractice.poc.domain.entity

import java.math.BigDecimal
import java.time.LocalDateTime

data class Order(
    val id: Int,
    val userId: Int,
    val userName: String,
    val productId: Int,
    val productName: String,
    val quantity: Int,
    val totalPrice: BigDecimal,
    val status: OrderStatus,
    val orderDate: LocalDateTime,
    val shippingAddress: String?
)

enum class OrderStatus {
    PENDING,
    CONFIRMED,
    SHIPPED,
    DELIVERED,
    CANCELLED
}

File: closet-ecommerce/closet-web/src/types/order.ts
export interface Order {
  id: number;
  orderNumber: string;
  memberId: number;
  items: OrderItem[];
  totalAmount: number;
  discountAmount: number;
  shippingFee: number;
  paymentAmount: number;
  status: OrderStatus;
  shippingAddress: string;
  shippingAddressDetail: string;
  recipientName: string;
  recipientPhone: string;
  zipCode: string;
  createdAt: string;
  updatedAt: string;
}

export interface OrderItem {
  id: number;
  productId: number;
  productName: string;
  productImage: string;
  optionName: string | null;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export enum OrderStatus {
  PENDING = 'PENDING',
  PAID = 'PAID',
  PREPARING = 'PREPARING',
  SHIPPING = 'SHIPPING',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED',
  REFUNDED = 'REFUNDED',
}

export interface CreateOrderRequest {
  items: CreateOrderItemRequest[];
  shippingAddressId: number;
  paymentMethod: string;
  couponId?: number;
}

export interface CreateOrderItemRequest {
  productId: number;
  optionId?: number;
  quantity: number;
}
